langgraph/src/checkpoint/sqlite.ts

```diff
@@ -0,0 +1,195 @@
+import Database, { Database as DatabaseType } from "better-sqlite3";
+import { RunnableConfig } from "@langchain/core/runnables";
+import {
+  BaseCheckpointSaver,
+  Checkpoint,
+  CheckpointAt,
+  CheckpointTuple,
+  SerializerProtocol,
+} from "./base.js";
+
+export class JsonSerializer implements SerializerProtocol {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  dumps(obj: any): string {
+    return JSON.stringify(obj);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  loads(data: string): any {
+    return JSON.parse(data);
+  }
+}
+
+// snake_case is used to match Python implementation
+interface Row {
+  checkpoint: string;
+  parent_ts: string;
+  thread_id?: string;
+  thread_ts?: string;
+}
+
+export class SqliteSaver extends BaseCheckpointSaver {
+  serde = new JsonSerializer();
+
+  db: DatabaseType;
+
+  constructor(
+    connStringOrLocalPath: string,
+    serde?: SerializerProtocol,
+    at?: CheckpointAt
+  ) {
+    super(serde, at);
+    this.db = new Database(connStringOrLocalPath);
+    this.setup().catch((error) =>
+      console.log("Error initializing SqliteSaver", error)
+    );
+  }
+
+  static fromConnString(connStringOrLocalPath: string): SqliteSaver {
+    return new SqliteSaver(connStringOrLocalPath);
+  }
+
+  private async setup(): Promise<void> {
+    try {
+      this.db.exec(`
+CREATE TABLE IF NOT EXISTS checkpoints (
```

Comment:
prob worth enabling wal mode? I know I haven't done that in py, prob should there too

Reply:
> I know I haven't done that in py, prob should there too

Added this to my To Do list for later.